tools/test/mbedtls-test-rewrite-branch-style

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python3
```

Comment:
Automated testing was not originally in scope. But manual testing was barely scaling, and hard to convey. So I've started to work on automated tests. So far my only coverage target was `test_updating_6863`, where I've tried to reproduce [the scenario where Manuel ran into problems](https://github.com/Mbed-TLS/mbedtls-docs/pull/71#issuecomment-1369598793). I could reproduce only one of the two problems.

I haven't really attempted to give decent feedback on failure (or inspection capabilities to ensure that we're testing what it looks like). To understand what's going on, try placing `import pdb; pdb.set_trace()` in a strategic place.

The test data comes partly from hard-coded shas and partly from references. This is messy. I guess it should be references only? But then there's the difficulty of how to reproduce those references. So far, I use hard-coded shas for inputs, but a manually inspected reference result conveyed via a tag in the repository.